library/psa_crypto_slot_management.c

```diff
@@ -200,7 +200,7 @@ psa_status_t psa_validate_persistent_key_parameters(
     }
     else
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
-    if( lifetime != PSA_KEY_LIFETIME_PERSISTENT )
+    if( ! PSA_KEY_LIFETIME_IS_PERSISTENT ( lifetime ) )
         return( PSA_ERROR_INVALID_ARGUMENT );
```

Comment:
Switching from a direct compare to using a macro to allow persistent vendor format keys to use this path as well.